### ISSUE TITLE
recovery_wait_secs feature for tf.train.MonitoredTrainingSession() similar to the one present in tf.train.SessionManager()

### ISSUE THREAD

**Issue by abishgj-rzt** [tensorflow/tensorflow#10679]
I ran the distributed model for a small data set and for very few epochs for some test run.

The chief worker started normally and finished training (training time was less than 30 seconds) but the other workers did not start.

Below log message has been displayed in all the workers other than chief:
`Waiting for model to be ready.  Ready_for_local_init_op:  Variables not initialized: <list_of_variables >`

Then I found the reason behind this in [this](https://stackoverflow.com/questions/42986653/distributed-tensorflow-not-running-some-workers/43007657#43007657) Stackoverflow answer.

A `recover_wait_secs` feature would make it easier rather handling that timeout period manually.

**Comment by aselle** [comment]
@ispirmustafa, could you have a look.

**Comment by tensorflowbutler** [comment]
It has been 14 days with no activity and this issue has an assignee.Please update the label and/or status accordingly.

**Comment by kdmcser** [comment]
I have the same problem. Waiting for 30s is too long for some test case.  I want to set it to 1s or below.

**Comment by ispirmustafa** [comment]
I agree it's a good idea to reveal 'recover_wait_secs` in `MonitoredTrainingSession`.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
A member of the TensorFlow organization has replied after the stat:awaiting tensorflower label was applied.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by tensorflowbutler** [comment]
Please remove the assignee, as this issue is inviting external contributions. Otherwise, remove the `contributions welcome` label. Thank you.

**Comment by marchss** [comment]
@abishgj-rzt  How did you handle the timeout now? I am also facing the same issue.

**Comment by sushreebarsa** [comment]
@abishgj-rzt One potential approach can be Customize ChiefSessionCreator. Could you try to do this by creating a custom ChiefSessionCreator subclass and within its create_session method, implement a retry loop with a wait time. If you still have a concern then please create a new ticket.
Thank you!
